### PR TITLE
Implement additional different voting power unit tests

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
Implement additional unit tests on PBFT which simulates different voting power scenarios:
1. There is quorum size of accumulated voting power of PREPARE messages, but no COMMIT messages => state machine remains in the `ValidateState`.
2. There is 2*F round change messages for greater round => state machine is in `AcceptState` with given round.

### CODE REVIEW

**Review by ferranbt — CHANGES_REQUESTED** [review]
Please split this into two PRs. One for the test and another one for the framework abstraction.

**Review by vcastellm — DISMISSED** [review]
LGTM

File: consensus_test.go [hunk 1]
```diff
@@ -489,17 +508,52 @@ func TestTransition_ValidateState_MoveToCommitState(t *testing.T) {
...
+		m.setState(ValidateState)
+		m.emitMsg(createMessage(NodeID("B"), MessageReq_Commit, nil))
+		m.emitMsg(createMessage(NodeID("C"), MessageReq_Commit, nil))
+
+		m.runCycle(context.Background())
+
+		m.expect(expectResult{
+			sequence:   1,
+			state:      RoundChangeState,
+			commitMsgs: 2, // Commit messages (A proposer sent commit via state machine loop, B sent commit via emit message)
```
Comment (ferranbt) [inline consensus_test.go:537]: We should add another condition here to check the aggregated voting power as well.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: consensus_test.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply consensus_test.go:537]: Agreed.

Comment (Stefan-Ethernal) [reply consensus_test.go]: Fixed https://github.com/0xPolygon/pbft-consensus/pull/73/commits/386a435ae2576f0cc14ccebbff9c5c9faf3d8502

**Review by ferranbt — APPROVED** [review]

**Pull request merged**